ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/AbstractJsonRpcHttpBySpecTest.java

```diff
@@ -29,22 +40,64 @@
 @RunWith(Parameterized.class)
 public abstract class AbstractJsonRpcHttpBySpecTest extends AbstractJsonRpcHttpServiceTest {
 
-  private final String specFileName;
+  private final URL specFile;
 
-  public AbstractJsonRpcHttpBySpecTest(final String specFileName) {
-    this.specFileName = specFileName;
+  public AbstractJsonRpcHttpBySpecTest(final String specName, final URL specURL) {
+    this.specFile = specURL;
   }
 
   @Test
   public void jsonRPCCallWithSpecFile() throws Exception {
-    jsonRPCCall(specFileName);
+    jsonRPCCall(specFile);
   }
 
-  private void jsonRPCCall(final String name) throws IOException {
-    final String testSpecFile = name + ".json";
-    final String json =
-        Resources.toString(
-            AbstractJsonRpcHttpBySpecTest.class.getResource(testSpecFile), Charsets.UTF_8);
+  /**
+   * Searches the provided subdirectories for spec files containing test cases, and returns
+   * parameters required to run these tests.
+   *
+   * @param subDirectoryPaths The subdirectories to search.
+   * @return Parameters for this test which will contain the name of the test and the url of the
+   *     spec file to run.
+   */
```

Comment:
getParams seems to be a confusing name.  I understand that it is going to a JUnit parameterized tests, but on it's own it's unclear.  Perhaps `findSpecTests`?